File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mpp = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(mpp.get(target-nums[i]) != null){
                return new int[]{mpp.get(target-nums[i]),i};
            }else{
                mpp.put(nums[i],i);
            }
        }
        return new int[]{-1,-1};
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        subset = []

        def dfs(i,target):
            if i == len(nums):
                if target == 0:
                    res.append(subset.copy())
                return
            
            if nums[i] <= target:
                subset.append(nums[i])
                dfs(i,target-nums[i])
                subset.pop()
            
            dfs(i+1,target)
        
        dfs(0,target)
        return res